velox/substrait/CMakeLists.txt

```diff
@@ -12,25 +12,56 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
-# Set up Proto
-set(proto_directory ${CMAKE_CURRENT_SOURCE_DIR}/proto)
-set(substrait_proto_directory ${CMAKE_CURRENT_SOURCE_DIR}/proto/substrait)
-set(PROTO_OUTPUT_DIR "${CMAKE_CURRENT_SOURCE_DIR}/proto/")
-file(MAKE_DIRECTORY ${CMAKE_CURRENT_SOURCE_DIR}/proto/substrait)
-file(GLOB PROTO_FILES ${substrait_proto_directory}/*.proto
-     ${substrait_proto_directory}/extensions/*.proto)
+# Download substrait proto files
+file(MAKE_DIRECTORY ${CMAKE_CURRENT_SOURCE_DIR}/proto)
+execute_process(
+  COMMAND git clone -c advice.detachedHead=false -b v0.23.0
+          https://github.com/substrait-io/substrait.git substrait-repo
+  WORKING_DIRECTORY ${CMAKE_CURRENT_SOURCE_DIR}/proto
+  RESULT_VARIABLE COMMAND_STATUS)
+if(COMMAND_STATUS EQUAL "1")
+  message(FATAL_ERROR "Failed to download substrait proto files")
+endif()
```

Comment:
Easier and better would be to use `FetchContent_*` as it is possible to use that to download things that don`t contain a CMakeLists.txt. That would also make it possible to replace the use of sed with `PATCH_COMMAND`.

```
FetchContent_Declare(substrait-proto 
...    
PATCH_COMMAND git apply .../substrait-proto.patch)


FetchContent_MakeAvailable(substrait-proto)
```
Afterwards `substrait-proto_SOURCE_DIR` is set to the root of the cloned and patched repo.

Reply:
Thanks for your suggestion, I will have a try!

Reply:
Done